### ISSUE TITLE
example6: Timeout expired on job submissions

### ISSUE THREAD

**Issue by cmoussa1** [flux-framework/flux-workflow-examples#20]
Walking through the steps on example 6 as is results in the following output when running `parent.sh`:

```
bash-4.2$ ./parent.sh 
Wed Oct 16 10:06:59 PDT 2019
submit: Submitted jobid 1
submit: Submitted jobid 2
submit: Submitted jobid 3
.
.
.
submit: Submitted jobid 213
submit: Submitted jobid 214
submit: Submitted jobid 215
submit: Submitted jobid 216
1000.051s: Timeout expired!
    ID NTASKS STATE                    START      RUNTIME    RANKS COMMAND
     6      1 running    2019-10-16T10:17:10       6.605m        2 flux
     5      1 running    2019-10-16T10:16:47       6.983m        1 flux
     4      1 running    2019-10-16T10:16:42       7.065m        0 flux
   216      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   215      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   214      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   213      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   212      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   211      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   210      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   209      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   208      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   207      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   206      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   205      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   204      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   203      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   202      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   201      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   200      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   199      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   198      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   197      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   196      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
   195      1 submitted  2019-10-16T10:23:46       0.000s      nil flux
First Level Done
```

Is "Timeout expired!" expected behavior?

**Comment by grondo** [comment]
Unlikely that is expected. The timeout is coming from `kvs-watch-until.lua -t ${MAXTIME} ${KEY} 'v == "complete"'`.

Something is wrong, since the example seems to submit `sleep 0` jobs, but jobs `4`, `5`, `6` have been running for minutes.

**Comment by dongahn** [comment]
This is a hierarchical nested scheduling demo. You are submitting 216 jobs each with 5000 sleep 0 jobs. So it is likely that this reached the slurm wall clock time.

A few things:

1. For `flux submit -N 1 -n 1 flux start -s1 ensemble.sh`
I wonder if you need `-s1` here. I don't think there is any harm, but `flux start` should get the size information from the allocation info. So maybe drop this option?

2. You probably want to change `NCORES` to be a smaller number. I think this meant to be adjusted to be the number of nodes times the number of cores per node. (e.g., 3 x 36)

3. I think we probably want to change:

`setenv FLUX_SCHED_OPTIONS "node-excl=true" `
to `setenv FLUX_SCHED_OPTIONS "node-excl=false"` 

so that the parent instance will do core-level scheduling for the outer loop

```
for i in `seq 1 ${NCORES}`; do
    flux submit -N 1 -n 1 flux start ensemble.sh
done
```

So all `NCORES` jobs can run simultaneously. 

4. You probably want to reduce the number of `sleep 0` jobs in the ensemble to something like 16. It can take a long time to schedule/execute 5000 jobs even if the runtime of each program is negligible.

**Comment by cmoussa1** [comment]
Going to close this issue - the fix and proper notes were made in PR #36
